src/systemctl/systemctl-util.c

```diff
@@ -624,25 +624,14 @@ static int unit_find_template_path(
         return r;
 }
 
-int unit_is_masked(sd_bus *bus, LookupPaths *lp, const char *name) {
+int unit_is_masked(sd_bus *bus, const char *unit) {
         _cleanup_free_ char *load_state = NULL;
         int r;
 
-        if (unit_name_is_valid(name, UNIT_NAME_TEMPLATE)) {
-                _cleanup_free_ char *path = NULL;
-
-                /* A template cannot be loaded, but it can be still masked, so
-                 * we need to use a different method. */
-
-                r = unit_file_find_path(lp, name, &path);
-                if (r < 0)
-                        return r;
-                if (r == 0)
-                        return false;
-                return null_or_empty_path(path);
-        }
```

Comment:
you cannot just drop this part without replacement. if a unit template `foo@.service` is not passed here we'd try to find a loaded Unit by that name, but that does not exist, since that's a template name, and not instantiated.

Hence special casing for templates *is* necessary. My complaint was just that instead of using a client side lookup we should use the server side lookup via GetUnitFileState() which is the call that actually is fine with templates passed in.